src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -0,0 +1,321 @@
+            internal ulong nswap;
+            internal ulong cnswap;
+            internal int exit_signal;
+            internal int processor;
+            internal uint rt_priority;
+            internal uint policy;
+            internal ulong delayacct_blkio_ticks;
+            internal ulong guest_time;
+            internal long cguest_time;
+        }
+
+        private static string GetStatFilePathForProcess(int pid)
+        {
+            return RootPath + pid.ToString(CultureInfo.InvariantCulture) + "/stat";
+        }
+
+        internal static string GetTaskDirectoryPathForProcess(int pid)
+        {
+            return RootPath + pid.ToString(CultureInfo.InvariantCulture) + "/task/";
+        }
+
+        private static string GetStatFilePathForThread(int pid, int tid)
+        {
+            // Perf note: Calling GetTaskDirectoryPathForProcess will allocate a string,
+            // which we then use in another Concat call to produce another string.  The straightforward alternative,
+            // though, since we have five input strings, is to use the string.Concat overload that takes a params array. 
+            // This results in allocating not only the params array but also a defensive copy inside of Concat,
+            // which means allocating two five-element arrays.  This two-string approach will result not only in fewer 
+            // allocations, but also typically in less memory allocated, and it's a bit more maintainable.
+            return GetTaskDirectoryPathForProcess(pid) + tid.ToString(CultureInfo.InvariantCulture) + "/stat";
+        }
+
+        internal static ParsedStat ReadStatFile(int pid)
+        {
+            string stat = File.ReadAllText(GetStatFilePathForProcess(pid));
+            return ParseStatFile(stat);
+        }
+
+        internal static ParsedStat ReadStatFile(int pid, int tid)
+        {
+            string stat = File.ReadAllText(GetStatFilePathForThread(pid, tid));
+            return ParseStatFile(stat);
+        }
+
+        private static ParsedStat ParseStatFile(string statFileContents)
+        {
+            ParsedStat results = default(ParsedStat);
+
+            int index = 0;
+            int i = 0;
+            for (; i < ExpectedStatFileParts && index < statFileContents.Length; i++)
+            {
+                string part = GetNextPart(statFileContents, ref index);
+                switch (i)
+                {
+                    case 0:
+                        results.pid = ParseInt32(part);
+                        break;
+                    case 1:
+                        string filename = part;
+                        if (filename.StartsWith("(") && filename.EndsWith(")"))
+                        {
+                            filename = filename.Substring(1, filename.Length - 2);
+                        }
+                        else
+                        {
+                            Debug.Fail("Unexpected comm format");
```

Comment:
What's the rationale for not failing in release here?  We have this problem all over the place.  Do we expect up stack stuff to be tolerant to garbage returned by this function?

Note we do this in the parsing logic later as well.